src/Parser.js

```diff
@@ -996,8 +996,11 @@ export default class Parser {
             }
             symbol = new ParseNode(symbols[this.mode][text].group,
                             text, this.mode, nucleus);
-        } else if (supportedCodepoint(text.charCodeAt(0)) &&
-            (this.mode === "text" || this.settings.unicodeTextInMathMode)) {
+        } else if ((this.settings.allowAllSymbols &&
+                    text.charCodeAt(0) >= 0x80) ||
+                   (supportedCodepoint(text.charCodeAt(0)) &&
+                    (this.mode === "text" ||
+                     this.settings.unicodeTextInMathMode))) {
```

Comment:
Should `allowAllSymbols` override `unicodeTextInMathMode`, i.e., supported unicode characters should be allowed in math mode even if `!unicodeTextInMathMode` when `allowAllSymbols`?

Reply:
I believe that is how it works now. Do you think it's counterintuitive? Another alternative would be to have `allowAllSymbols` take an argument of `"math"` or `"text"`, and `true` for both. Whatever we decide, the behavior should be documented.

Reply:
@edemaine I'm not sure. But as it's more counterintuitive to disallow _supported_ characters in math mode while allowing _unsupported_ characters, I think this PR's behavior is OK.

Reply:
I added documentation that this is the behavior.  Should be good to go now.

I think this issue would be cleaner if we moved to a default `true` value of `unicodeTextInMathMode`, or default "nonstrict" from #1226.

Reply:
We should probably only have one setting so that people don't get confused about which setting to use or what the interaction between these two settings should be.  I think we should probably just introduce the `strict` setting in this diff in order to minimize the number of breaking changes.  We can transition the other settings to use the `strict` setting in a subsequent PR.